cognee/infrastructure/databases/vector/neptune_analytics/NeptuneAnalyticsAdapter.py

```diff
@@ -194,7 +194,38 @@ async def search(
             A list of scored results that match the query.
 
         """
-        pass
+        if query_vector and query_text:
+            raise InvalidValueError(
+                message="The search function accepts either text or embedding as input, but not both."
+            )
+        elif query_text is None and query_vector is None:
+            raise InvalidValueError(message="One of query_text or query_vector must be provided!")
+        elif query_vector:
+            embedding = query_vector
+        else:
+            data_vectors = (await self.embedding_engine.embed_text([query_text]))
+            embedding = data_vectors[0]
+
+        # Composite the parameters map
+        params = dict(embedding=embedding, param_topk=limit)
+        # Composite the query
+        query_string = f"""
+        CALL neptune.algo.vectors.topKByEmbeddingWithFiltering({{
+                topK: {limit},
+                embedding: {embedding}, 
+                nodeFilter: {{ equals: {{property: '{self.COLLECTION_PREFIX}', value: '{collection_name}'}} }}
+              }}
+            )
+        YIELD node, score
+        RETURN id(node) as id, node as payload, score as score
+        """
+        # Print the result
+        query_response = self._client.query(query_string, params)
+        return [ScoredResult(
+            id=item.get('payload').get('~id'),
```

Comment:
```suggestion
            id=item.get('id'),
```
or remove it from the query result?